py_src/ipyelk/diagram/layout_options/edge_options.py

```diff
@@ -0,0 +1,370 @@
+# Copyright (c) 2020 Dane Freeman.
+# Distributed under the terms of the Modified BSD License.
+
+from typing import List
+
+import ipywidgets as W
+import traitlets as T
+
+from ..elk_model import ElkEdge, ElkLabel
+from .selection_widgets import LayoutOptionWidget, SpacingOptionWidget
+
+EDGE_LABEL_OPTIONS = {
+    "Center": "CENTER",
+    "Head": "HEAD",
+    "Tail": "TAIL",
+}
+
+
+EDGE_TYPE_OPTIONS = {
+    "None": "NONE",
+    "Directed": "DIRECTED",
+    "Undirected": "UNDIRECTED",
+    "Association": "ASSOCIATION",
+    "Generalization": "GENERALIZATION",
+    "Dependency": "DEPENDENCY",
+}
+
+
+EDGESIDE_OPTIONS = {
+    "Alway Up": "ALWAYS_UP",
+    "Always Down": "ALWAYS_DOWN",
+    "Direction Down": "DIRECTION_DOWN",
+    "Smart Up": "SMART_UP",
+    "Smart Down": "SMART_DOWN",
+}
+
+
+EDGE_LABEL_PLACEMENT_STRATEGY = {
+    "Median Layer": "MEDIAN_LAYER",
+    "Tail Layer": "TAIL_LAYER",
+    "Head Layer": "HEAD_LAYER",
+    "Space Efficient Layer": "SPACE_EFFICIENT_LAYER",
+    "Widest Layer": "WIDEST_LAYER",
+    "Center Layer": "CENTER_LAYER",
+}
+
+
+EDGE_ROUTING_OPTIONS = {
+    "Undefined": "UNDEFINED",
+    "Polyline": "POLYLINE",
+    "Orthogonal": "ORTHOGONAL",
+    "Splines": "SPLINES",
+}
+
+
+class InlineEdgeLabels(LayoutOptionWidget):
+    """If true, an edge label is placed directly on its edge. May only apply to
+    center edge labels. This kind of label placement is only advisable if the
+    label’s rendering is such that it is not crossed by its edge and thus stays
+    legible.
+
+    https://www.eclipse.org/elk/reference/options/org-eclipse-elk-edgelabels-inline.html
+    """
+
+    identifier = "org.eclipse.elk.edgeLabels.inline"
+    applies_to = ElkLabel
+
+    group = "edgeLabels"
+
+    inline = T.Bool(default_value=False)
+
+    def _ui(self) -> List[W.Widget]:
+        cb = W.Checkbox(description="Inline Edge Labels")
```

Comment:
Perhaps we hoist the `description` to a static on the class... or standardize on
```
class FooBarBaz(LayoutOptionWidget):
  """ Foo Bar Baz
        first you foo, then you bar in order to baz
  """
   def _ui(self) -> List[W.Widget]:
        cb = W.Checkbox(description=self.__doc__.splitlines()[0].trim())
```